static-exporter/main.libsonnet

```
@@ -35,7 +35,7 @@ local k = import 'ksonnet-util/kausal.libsonnet';
             function(acc, metric)
               acc + [
                 '# HELP %(name)s %(description)s' % metric,
-                '# TYPE %(name)s counter' % metric,
+                '# TYPE %(name)s %(metricType)s' % metric,
```

This is string formatting with the `metric` object. 

The `self.with<FUNCTION>` indirection can make it a little hard to think about but the `metric` object has three fields, `name`, `description`, and `type`:

```bash
jsonnet -e "(import './main.libsonnet').metric.new('NAME', 'DESCRIPTION', 'TYPE')"

{
   "description": "DESCRIPTION",
   "name": "NAME",
   "type": "TYPE"
}
```

So I think you want:

```suggestion
                '# TYPE %(name)s %(type)s' % metric,
```